File: review/project-fe/src/app/model/customer/customer.ts
import {Address} from '../address/address';

export interface Customer {
  id?: number;
  name?: string;
  code?: string;
  phoneNumber?: string;
  dateOfBirth?: string;
  email?: string;
  gender?: boolean;
  idCard?: string;
  imgUrl?: string;
  status?: boolean;
  address?: Address;
}

File: pawn-shop/src/main/java/com/pawn_shop/model/customer/Customer.java
package com.pawn_shop.model.customer;

import com.pawn_shop.model.address.Address;
import com.pawn_shop.model.contract.Contract;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDate;
import java.util.Set;

@Getter
@Setter
@Entity
@AllArgsConstructor
@NoArgsConstructor
public class Customer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private String code;

    private String phoneNumber;

    private String email;

    private LocalDate dateOfBirth;

    private Boolean gender;

    private String idCard;

    private String imgUrl;

    private Boolean status;

    @OneToOne(mappedBy = "customer")
    private Address address;

    @OneToMany(mappedBy ="customer",cascade = CascadeType.ALL)
    private Set<Contract> contracts;
}
